lib/WUI/nhttp/gcode_upload.cpp

```diff
@@ -47,16 +47,29 @@ GcodeUpload::GcodeUpload(UploadState uploader, bool json_errors, size_t length,
     , file_idx(upload_idx) {
 }
 
+GcodeUpload::GcodeUpload(bool json_errors, size_t length, size_t upload_idx, unique_file_ptr file, UploadedNotify *uploaded, const char* filepath_other, bool print_after_upload)
+    : uploaded_notify(uploaded)
+    , size_rest(length)
+    , json_errors(json_errors)
+    , cleanup_temp_file(true)
+    , tmp_upload_file(move(file))
+    , file_idx(upload_idx)
+    , print_after_upload(print_after_upload){
+    strcpy(this->filepath, filepath_other);
```

Comment:
According to the header file, the `filepath_other` may be `nullptr`. I don't think that's valid.